File: ios/ModuleWithEmitter.m
//
//  ModuleWithEmitter.m
//  RocketChatRN
//
//  Created by Arti Mane on 12/02/21.
//  Copyright © 2021 Facebook. All rights reserved.
//


#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>


@interface RCT_EXTERN_MODULE(ModuleWithEmitter, RCTEventEmitter)
  RCT_EXTERN_METHOD(supportedEvents)
@end

File: ios/ModuleWithEmitter.swift
//
//  ModuleWithEmitter.swift
//  RocketChatRN
//
//  Created by Arti Mane on 12/02/21.
//  Copyright © 2021 Facebook. All rights reserved.
//

import Foundation
@objc(ModuleWithEmitter)
open class ModuleWithEmitter: RCTEventEmitter {

  public static var emitter: RCTEventEmitter!

  override init() {
    super.init()
    ModuleWithEmitter.emitter = self
  }

  open override func supportedEvents() -> [String] {
    ["onSessionConnect", "onPending", "onFailure","getInboundCall","VoipCall","testCall"]
  }
}
